packages/tools/src/tools/annotation/UltrasoundDirectionalTool.ts

```diff
@@ -0,0 +1,894 @@
+      };
+    }
+
+    return renderStatus;
+  };
+
+  _calculateCachedStats(annotation, renderingEngine, enabledElement) {
+    const data = annotation.data;
+    const { viewportId, renderingEngineId } = enabledElement;
+
+    // Until we have all two anchors bail out
+    if (data.handles.points.length !== 2) {
+      return;
+    }
+
+    const { cachedStats } = data;
+    const targetIds = Object.keys(cachedStats);
+
+    for (let i = 0; i < targetIds.length; i++) {
+      const targetId = targetIds[i];
+
+      const image = this.getTargetIdImage(targetId, renderingEngine);
+
+      // If image does not exists for the targetId, skip. This can be due
+      // to various reasons such as if the target was a volumeViewport, and
+      // the volumeViewport has been decached in the meantime.
+      if (!image) {
+        continue;
+      }
+
+      const { imageData } = image;
+
+      const worldPos1 = data.handles.points[0];
+      const worldPos2 = data.handles.points[1];
+
+      const imageIndex1 = transformWorldToIndex(imageData, worldPos1);
+      const imageIndex2 = transformWorldToIndex(imageData, worldPos2);
+
+      const { values: values1, units: units1 } =
+        getCalibratedProbeUnitsAndValue(image, [imageIndex1]);
+      const { values: values2, units: units2 } =
+        getCalibratedProbeUnitsAndValue(image, [imageIndex2]);
+
+      let value;
+      let unit = 'px';
+      if (
+        units1[0] !== units2[0] ||
+        units1[1] !== units2[1] ||
+        (units1[0] === 'raw' && units2[0] === 'raw')
+      ) {
+        // if units are not the same, we cannot calculate the diff
+        // so we just report the px distance
+        value = distanceToPoint(worldPos1, worldPos2);
+      } else {
+        const canvasPoint1 = enabledElement.viewport.worldToCanvas(worldPos1);
+        const canvasPoint2 = enabledElement.viewport.worldToCanvas(worldPos2);
+
+        const canvasDeltaY = canvasPoint2[1] - canvasPoint1[1];
+        const canvasDeltaX = canvasPoint2[0] - canvasPoint1[0];
+
+        const orientation =
+          Math.abs(canvasDeltaX) > Math.abs(canvasDeltaY)
+            ? 'HORIZONTAL'
+            : 'VERTICAL';
+
+        // then for the third point we need to go from first point towards
+        // the second point (it can be left or right in the horizontal orientation)
+        // or up or down in the vertical orientation, and only add
+        // the delta y to the x or y coordinate of the first point
+        const xValues = [values1[0], values2[0]];
+        const yValues = [values1[1], values2[1]];
+
+        if (orientation === 'HORIZONTAL') {
+          value = Math.abs(xValues[1] - xValues[0]);
+          unit = units1[0];
+        } else if (orientation === 'VERTICAL') {
+          value = Math.abs(yValues[1] - yValues[0]);
+          unit = units1[1];
+        } else {
+          throw new Error('Invalid orientation');
+        }
+      }
+
+      cachedStats[targetId] = {
```

Comment:
If the cached stats has a set of indices for the values to display, then you can easily add the ability to display both - so you have two orientations and three sets of values to display:
horizontal/vertical orientation
both values, eg [0,1]
horizontal value [1]
vertical value [0]
That can be in the configuration for the tool as a whole, so that you can configure the same tool to always display both, regardless of orientation, or only display one or the other regardless of orientation, but put the label in the same spot.

Reply:
Ok done